controller/src/main/java/org/jboss/as/controller/operations/validation/EnumValidator.java

```diff
@@ -136,13 +136,16 @@ public EnumValidator(final Class<E> enumType,  final boolean nullable, final boo
      * Creates a new validator for the enum type with the allowed values defined in the {@code allowed} parameter.
      *
      * @param enumType the type of the enum.
-     * @param allowed  the enum values that are allowed.
+     * @param allowed  the enum values that are allowed. If {@code null} or zero length this is interpreted as meaning all values
      * @param <E>      the type of the enum.
      *
      * @return a new validator.
      */
     @SafeVarargs
     public static <E extends Enum<E>> EnumValidator<E> create(final Class<E> enumType, final E... allowed) {
+        if (allowed == null || allowed.length == 0) {
+            return create(enumType, EnumSet.allOf(enumType));
+        }
```

Comment:
Note that this is a behavior change in this method, as before providing a null or empty array would result in a validator that would reject all values.  That is non-sensical behavior and I checked wf-core and wf code for any uses that passed null or empty and found none, so I made this change.